### DISCUSSION TITLE
Issue connecting to Azure Postgres DataBase Netbox v3.6.2

### DISCUSSION THREAD

**Discussion by AzRe3aL** [netbox-community/netbox-docker#1088] [Getting Help]
Updated from v3.5 > 3.6.3

The container starts up and runs through start-up scripts. when attempting to connect to netbox webUI (Error 503) when looking at the logs I see 

django.db.utils.OperationalError: connection failed: could not open certificate file "/root/.postgresql/postgresql.crt": Permission denied

connection to server at "MY AZURE DATABASE" (x.x.x.x), port 5432 failed: FATAL:  no pg_hba.conf entry for host "x.x.x.x", user "xxxxxxxxxxx", database "xxxxxxxxxxx", SSL off
x.x.x.x - - [21/Sep/2023:14:32:03 +0000] "GET / HTTP/1.1" 503 54 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36"
2023/09/21 14:34:04 [notice] 20#20 app process 21 exited with code 0

Any help would be greatly appreciated

**Comment by HungryHowies** [comment]
Hey @AzRe3aL 

This  error...

"django.db.utils.OperationalError: connection failed: could not open certificate file "/root/.postgresql/postgresql.crt": Permission denied"

 Try setting your certificates to somewhere that  Netbox  can access them. Or set the persions on the directory 

`/root/.postgresql/postgresql.crt`

That would be my guess.

**Reply by es-cs1** [reply]
I see a similar issue here
https://github.com/netbox-community/netbox-docker/issues/1112

So going forward with all future releases of netbox, are we expected to provide a self-signed client .crt and .key file on our netbox instance to connect against the Postgresql Server?

If so would this want to be included in the docs somewhere?

**Comment by marsteel** [comment]
> Updated from v3.5 > 3.6.3
> 
> The container starts up and runs through start-up scripts. when attempting to connect to netbox webUI (Error 503) when looking at the logs I see
> 
> django.db.utils.OperationalError: connection failed: could not open certificate file "/root/.postgresql/postgresql.crt": Permission denied
> 
> connection to server at "MY AZURE DATABASE" (x.x.x.x), port 5432 failed: FATAL: no pg_hba.conf entry for host "x.x.x.x", user "xxxxxxxxxxx", database "xxxxxxxxxxx", SSL off x.x.x.x - - [21/Sep/2023:14:32:03 +0000] "GET / HTTP/1.1" 503 54 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36" 2023/09/21 14:34:04 [notice] 20#20 app process 21 exited with code 0
> 
> Any help would be greatly appreciated

Did you find how to fix the issue or any workaround? Thanks.

**Comment by marsteel** [comment]
Here is my workaround. 

My Dockerfile

``
```dockerfile
FROM netboxcommunity/netbox:v3.7.6-2.8.0

USER root
RUN curl -LJO https://truststore.pki.rds.amazonaws.com/eu-west-1/eu-west-1-bundle.pem -o /tmp/eu-west-1-bundle.pem
RUN mkdir -p /root/.postgresql
RUN curl  https://truststore.pki.rds.amazonaws.com/eu-west-1/eu-west-1-bundle.pem -o /root/.postgresql/postgresql.crt
RUN chmod -R 777 /root
``` 

For PEM files in other regions, https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html#UsingWithRDS.SSL.RegionCertificates

**Reply by AzRe3aL** [reply]
We resolved the issue, however, I have completely forgot how, I would need to look back and see if I can work out how we fix it, sorry

**Comment by blindzero** [comment]
I am wondering about this issue as well.
I am using PgSQL on Azure and connection from docker is failing for same reason. The error message of missing pg_hba.conf entry is misleading as netbox seems to fall back to "no encryption" which is not supported at all.
However, none of my other applications had issue to resolve the Azure certificate as it is usually included in CA packages...?
